gapis/resolve/dependencygraph2/graph_visualization/graph_algorithms.go

```diff
@@ -152,3 +158,110 @@ func (g *graph) joinNodesWithZeroDegree() {
 		}
 	}
 }
+
+func (g *graph) assignColorToNodes() {
+	for _, currentNode := range g.nodeIdToNode {
+		currentNode.color = currentNode.label.GetTopLevelName()
+	}
+}
+
+func (g *graph) joinNodesThatDoNotBelongToAnyFrame() {
+	for _, currentNode := range g.nodeIdToNode {
+		if currentNode.label.GetSize() > 0 && currentNode.label.GetTopLevelName() != FRAME {
+			currentNode.label.PushFront(UNUSED, 0)
+		}
+	}
+}
+
+type chunk struct {
+	idToPosition       map[int]int
+	positionToLevelsID [][]int
+	done               bool
+}
+
+func assignLevelsIDToChunk(leftPosition, rightPosition int, currentLevelsID *[]int, positionToLevelsID *[][]int) {
+	if (rightPosition - leftPosition + 1) <= MAXIMUM_NUMBER_OF_NODES_BY_LEVEL {
+		for i := leftPosition; i <= rightPosition; i++ {
+			(*positionToLevelsID)[i] = make([]int, len(*currentLevelsID))
+			copy((*positionToLevelsID)[i], *currentLevelsID)
+		}
+	} else {
+		*currentLevelsID = append(*currentLevelsID, 1)
+		chunkSize := (rightPosition - leftPosition + 1 + MAXIMUM_NUMBER_OF_NODES_BY_LEVEL - 1) / MAXIMUM_NUMBER_OF_NODES_BY_LEVEL
+		newLeftPosition := leftPosition
+		newRightPosition := newLeftPosition + chunkSize - 1
+		chunkId := 1
+		for newLeftPosition <= rightPosition {
+			if newRightPosition > rightPosition {
+				newRightPosition = rightPosition
+			}
+			(*currentLevelsID)[len(*currentLevelsID)-1] = chunkId
+			assignLevelsIDToChunk(newLeftPosition, newRightPosition, currentLevelsID, positionToLevelsID)
+			newLeftPosition = newRightPosition + 1
+			newRightPosition = newLeftPosition + chunkSize - 1
+			chunkId++
+		}
+		*currentLevelsID = (*currentLevelsID)[:len(*currentLevelsID)-1]
+	}
+}
+
+func makeChunks(nodes []*node) {
```

Comment:
I think this function could use a comment describing its purpose.

Reply:
I added a comment to describe the purpose, also I renamed some variables to better understanding of the code.